client/src/components/NicknameDropdown.jsx

```diff
@@ -4,19 +4,37 @@
  * Part of Silesian University of Technology project.
  * Created only for learning purposes.
  */
+import { useState } from 'react';
 import PropTypes from 'prop-types';
+import { UniqueOTP } from 'unique-string-generator';
 import { useAppContext } from '../context/AppContextProvider';
 import CustomButton from './CustomButton';
 
 const NicknameDropdown = ({ nicknameInput, handleSave }) => {
-  const { state } = useAppContext();
+  const { state, dispatch } = useAppContext();
+  const [tempNickname, setTempNickname] = useState(state.nickname);
 
   const handleKeyDown = event => {
     if (event.key === 'Enter') {
-      handleSave();
+      saveNickname();
     }
   };
 
+  const setDefaultNickname = () => {
+    const defaultNickname = `Guest${UniqueOTP(5)}`;
+    setTempNickname(defaultNickname);
+  };
+
+  const saveNickname = () => {
+    localStorage.setItem('nickname', tempNickname);
+    dispatch({ type: 'setNickname', value: tempNickname });
+    handleSave();
+  };
+
+  const handleChange = event => {
+    setTempNickname(event.target.value);
+  };
+
```

Comment:
Please move creating `defaultNickname` into util function (also replace in AppContext).